### PR TITLE
Support arithmetic predicate pushdown for Phoenix

### PR DESCRIPTION
**Pull request by chenjian2664**
## Description
https://github.com/trinodb/trino/issues/16732

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Phoenix
* Support arithmetic predicate pushdown. ({issue}`16732`)
```

### CODE REVIEW

File: plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java [hunk 1]
```diff
@@ -87,6 +87,9 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
             case SUPPORTS_AGGREGATION_PUSHDOWN:
                 return false;
 
+            case SUPPORTS_PREDICATE_ARITHMETIC_EXPRESSION_PUSHDOWN:
```
[the remark this answers is not in the document]
Comment (chenjian2664) [reply plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java:90]: There is one test `BaseJdbcConnectorTest.testArithmeticPredicatePushdown` for this, any thoughts?

[the remark this answers is not in the document]
Comment (hashhar) [reply plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java:90]: the current one only includes modulus tests. I was adding more in https://github.com/trinodb/trino/pull/14939 but got stalled.

Can you copy the tests from the linked PR to Phoenix for now and when I get back to that PR I can remove the extra ones in Phoenix?

File: plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java [hunk 2]
```diff
@@ -117,6 +120,100 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
...
+                .isFullyPushedDown();
+        assertThat(query("SELECT orderkey, custkey FROM orders WHERE custkey = 1483 AND orderkey % custkey = 1470"))
+                .isFullyPushedDown();
+
+        // some databases calculate remainder instead of modulus when one of the values is negative
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % -nationkey = 2"))
+                .isFullyPushedDown();
+    }
+
+    protected void verifyDivisionByZeroFailure(Throwable e)
```
Comment (hashhar) [inline plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java:212]: inline since it's the only usage in this class

**Review by hashhar — APPROVED** [review]
LGTM % comment

Thanks a lot.

File: plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java (hunk 2 above)
Comment (chenjian2664) [reply plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java:212]: Do you mean change it to `private` method or define a `Consumer` inside the test `testArithmeticPredicatePushdown` instead?

Comment (hashhar) [reply plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java:212]: Just make it private I think.

**Comment by chenjian2664** [comment]
Thanks for your time for review it @hashhar

**Pull request merged**